### PR TITLE
MOSIP-15993: Upgrading swagger2 to openapi3.0 in registration processor 

### PR DESCRIPTION
**Pull request by kameshsr**
Successfully generated openapi3.0 after upgrading the version of swagger and some related annotations like @Apioperation to @Operation and @Api to @Tag.

### CODE REVIEW

File: registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/controller/RegistrationExternalStatusController.java [hunk 1]
```diff
@@ -82,7 +72,7 @@
 	 */
 	@PreAuthorize("hasAnyRole('REGISTRATION_ADMIN', 'REGISTRATION_OFFICER', 'REGISTRATION_SUPERVISOR','RESIDENT')")
 	@PostMapping(path = "/externalstatus/search", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-	@ApiOperation(value = "Get the registration external status", response = RegistrationExternalStatusCode.class)
+	@Operation(summary = "Get the registration external status", description = "registration external status", tags = { "External Registration Status" })
```
Comment (vishwa-vyom) [inline registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/controller/RegistrationExternalStatusController.java]: For all APIs have the summary and description as same for now

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/config/RegistrationStatusConfig.java [hunk 2]
```diff
@@ -1,31 +1,26 @@
...
 	 */
 	@Bean
-	public Docket registrationStatusBean() {
-		return new Docket(DocumentationType.SWAGGER_2).groupName("Registration Status").select()
-				.apis(RequestHandlerSelectors.basePackage("io.mosip.registration.processor.status.api.controller"))
-				.paths(PathSelectors.ant("/*")).build();
+	public OpenAPI customOpenAPI() {
+		return new OpenAPI()
+				.components(new Components())
+				.info(new Info().title("Registration status API").description(
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/config/RegistrationStatusConfig.java]: done

File: registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/config/RegistrationStatusConfig.java [hunk 3]
```diff
@@ -1,31 +1,26 @@
...
+						"This is a registration status service using springdoc-openapi and OpenAPI 3.").version("3.0.1"));
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/config/RegistrationStatusConfig.java]: Yes sir updated

File: registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/controller/RegistrationExternalStatusController.java (hunk 1 above)
Comment (kameshsr) [reply registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/controller/RegistrationExternalStatusController.java]: Yes, I changed all summaries and descriptions.

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**